mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java

```diff
@@ -288,4 +288,12 @@ private static InputStream getMessagingDescriptorInputStream() throws MessagingD
   public static String getModuleId() {
     return format("%s-%s", PomReader.INSTANCE.getModuleName(), PomReader.INSTANCE.getVersion());
   }
+
+  private static PubsubClient createPubSubClient(OkapiConnectionParams params) {
+    if (params.getVertx() == null) {
+      throw new NullPointerException("The vertex object is required, should not be null!");
```

Comment:
I don't think that explicitly throwing an exception here is a good idea. Can we return a failed future instead?

Reply:
I would remove that code fragment entirely. It will throw a runtime exception anyway. And Vert.x will catch it.

Reply:
done